website/versioned_docs/version-5.x/ecosystem-snowpack.md

```diff
@@ -0,0 +1,102 @@
+---
+id: ecosystem-snowpack
+title: Snowpack
+sidebar_label: Snowpack
+---
+
+[Snowpack](https://www.snowpack.dev/) is a tool for both local development and the building of applications. It uses in-browser ES modules during development, and then bundles with webpack (or other build tools) for production.
+
+## Example repo
+
+https://github.com/joeldenning/snowpack-single-spa-example
+
+## Overview
+
+Snowpack uses ES modules in local development, but not in production. This works well with single-spa, which encourages using [in-browser modules](/docs/recommended-setup#in-browser-versus-build-time-modules) as the interface for each microfrontend. To use snowpack with single-spa, you must export the [single-spa lifecycle functions](/docs/building-applications#registered-application-lifecycle) from your Snowpack project's `index.js` file and then make a few modifications to the `snowpack.config.js` file.
+
+## Configuration
+
+Modify the `index.js` file to not mount your app immediately, but rather to export the single-spa lifecycles. If using Vue, for example, see https://single-spa.js.org/docs/ecosystem-vue#usage.
+
+The following Snowpack config can be used as the basis for a single-spa + Snowpack setup. It requires installing [systemjs-webpack-interop](https://github.com/joeldenning/systemjs-webpack-interop) and `@snowpack/plugin-webpack`:
+
+```sh
+npm install --save-dev systemjs-webpack-interop @snowpack/plugin-webpack
+
+yarn add --dev systemjs-webpack-interop @snowpack/plugin-webpack
+
+pnpm install --save-dev systemjs-webpack-interop @snowpack/plugin-webpack
+```
+
+```js
+const { merge } = require("webpack-merge");
+const SystemJSPublicPathWebpackPlugin = require("systemjs-webpack-interop/SystemJSPublicPathWebpackPlugin");
+
+/** @type {import("snowpack").SnowpackUserConfig } */
+module.exports = {
+  mount: {
+    /* ... */
+  },
+  plugins: [
+    [
+      "@snowpack/plugin-webpack",
+      {
+        extendConfig(config) {
+          delete config.optimization.runtimeChunk;
+          delete config.optimization.splitChunks;
+
+          return merge(config, {
+            mode: "development",
+            module: {
+              rules: [
+                {
+                  parser: {
+                    system: false,
```

Comment:
Would it be worth noting that this is only compatible with whatever versions of `@snowpack/plugin-webpack` use Webpack 4?

Reply:
There are no versions of `@snowpack/plugin-webpack` that use webpack 5 yet, as far as I know. See https://github.com/snowpackjs/snowpack/blob/41c29f25e53695ec84c07386af7eb140d0b22a8e/plugins/plugin-webpack/package.json#L28

Reply:
Right, I'm just thinking of if/when they do, it might trip people up. But maybe we can add that once such a version does exist.

Reply:
I'll put a comment in now